server/controllers/auth.js

```diff
@@ -52,3 +55,75 @@ exports.login = async (req, res, next) => {
     }
   }
 };
+
+exports.registerController = async (req, res, next) => {
+  try {
+    const {
+      firstName,
+      lastName,
+      password,
+      email,
+      role,
+    } = await registerSchema.validate(req.body, {
+      abortEarly: false,
+    });
+    let existingUser;
+    switch (role) {
+      case 'artist':
+        existingUser = await getArtistByEmail(email);
+        break;
+
+      case 'customer':
+        existingUser = await checkCustomerEmail(email);
+        break;
+      default:
+        throw new Error('Choose your role');
+    }
+    if (existingUser.rows.length !== 0)
+      return res
+        .status(400)
+        .json({ statusCode: 400, message: 'email exist !!' });
+
+    let newPassword;
+    try {
+      const hashPass = await hash(password, 10);
```

Comment:
You should use ` gensalt() ` method to get salt  which   takes an optional parameter ` (log rounds) ` that determines the computational complexity of the hashing
```suggestion
      const salt = await genSalt(10);
      const hashPass = await hash(password, salt);
```